packages/react/menu/src/Menu.tsx

```diff
@@ -710,8 +743,31 @@ const MenuSubTrigger = React.forwardRef((props, forwardedRef) => {
         data-state={getOpenState(context.open)}
         {...props}
         ref={composeRefs(forwardedRef, context.onTriggerChange)}
-        onMouseMove={composeEventHandlers(props.onMouseMove, () => {
-          if (!props.disabled) context.onOpenChange(true);
+        onMouseMove={composeEventHandlers(props.onMouseMove, (event) => {
+          if (contentContext.isMouseMovingToSubmenu(event)) return;
+          if (!props.disabled && !context.open) {
+            openTimersRef.current.push(window.setTimeout(() => context.onOpenChange(true), 100));
```

Comment:
I'm really struggling to work out why we need an array of open timers, what's the reasoning?

Reply:
It's because it's in `onMouseMove`, so in `onMouseLeave` when we want to clear the timeout, we'd be clearing only the last one, but there's 100ms where new timeouts would override it and thus old ones would be lost and un-clearable. Let me know if that makes sense, we can go on a call otherwise.

Reply:
Yep makes sense, can we add that as a comment 🙏 ?